proc_macros/src/helpers.rs

```diff
@@ -0,0 +1,46 @@
+use std::collections::HashMap;
+
+use syn::{
+    parse::{Parse, ParseStream},
+    Expr, Ident, Token,
+};
+
+pub struct ArrowSeparatedKeyValuePair<TKey = Ident, TValue = Expr> {
+    key: TKey,
+    value: TValue,
+}
+
+impl<TKey, TValue> Parse for ArrowSeparatedKeyValuePair<TKey, TValue>
+where
+    TKey: Parse,
+    TValue: Parse,
```

Comment:
Nice! I wonder if anyone has done stuff like this in a library. It seems like it'd come up quite a bit.

Reply:
Ya I'm intrigued when I see people who have bothered to try and abstract out "proc macro helpers/"building blocks"" (seems like an "itch that people scratch" who are writing a decent amount of proc macros as part of their coding style) but I'm not really familiar with "what's out there"